File: update_sample_csv.py
import json
import csv
from pathlib import Path

def main():
    data_file = Path('data/clients.json')
    out_file = Path('data/sample_clients.csv')
    if not data_file.exists():
        raise SystemExit(f"Missing {data_file}")
    with data_file.open() as f:
        clients = json.load(f)
    # Write CSV with header: access_code,client_name
    with out_file.open('w', newline='') as f:
        writer = csv.writer(f)
        writer.writerow(['access_code', 'client_name'])
        for code, info in clients.items():
            writer.writerow([code, info.get('clientName', '')])
    print(f"Written {len(clients)} clients to {out_file}")

if __name__ == '__main__':
    main()

File: scripts/update_sample_csv.js
const fs = require('fs');
const path = require('path');

const clientsPath = path.join(__dirname, '..', 'data', 'clients.json');
const csvPath = path.join(__dirname, '..', 'data', 'clients_sample.csv');

const clients = JSON.parse(fs.readFileSync(clientsPath, 'utf8'));
const rows = ['accessCode,clientName'];
for (const [code, { clientName }] of Object.entries(clients)) {
  rows.push(`${code},${clientName}`);
}
fs.writeFileSync(csvPath, rows.join('\n'));
console.log(`CSV updated at ${csvPath}`);
